src/viewability/ViewabilityHelper.ts

```diff
@@ -138,7 +138,8 @@ class ViewabilityHelper {
       Math.min(itemTop + itemSize, listMainSize) - Math.max(itemTop, 0);
 
     // Always consider item fully viewable if it is fully visible, regardless of the `viewAreaCoveragePercentThreshold`
-    if (pixelsVisible === itemSize) {
+    // Account for floating point imprecision.
+    if (Math.abs(pixelsVisible - itemSize) <= 0.001) {
```

Comment:
We could also do [what FlatList is doing](https://github.com/facebook/react-native/blob/main/packages/virtualized-lists/Lists/ViewabilityHelper.js#L344-L350) to implement this check.

```TypeScript
const itemBottom = itemTop + itemSize;
if (itemTop >= 0 && itemBottom <= listMainSize && itemBottom > itemTop) {
    return true;
}
```